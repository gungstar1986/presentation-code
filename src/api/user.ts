import { User } from 'store/user/interfaces';

export const apiUser = {
  userAuth: (user: User) => {
    try {
      return { ...user };
    } catch (e) {
      throw e;
    }
  },
};
