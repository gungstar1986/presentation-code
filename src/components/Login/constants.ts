import * as Yup from 'yup';

import { LoginData } from 'components/Login/interfaces';

export const loginFormikData: LoginData = {
  email: '',
  password: '',
  rememberMe: false,
};

const passValidate = {
  numbers: /(?=.*[0-9])/,
  lowercase: /(?=.*[a-z])/,
  uppercase: /(?=.*[A-Z])/,
  symbols: /(?=.*[*@#$%!&])/,
  length: /[0-9a-zA-Z*@#$%!&]{8,}/,
};

export const loginFormicSchema = Yup.object({
  email: Yup.string().email('Incorrect field').required('Required field'),
  password: Yup.string()
    .matches(passValidate.lowercase, 'At least 1 symbol in lower case')
    .matches(passValidate.uppercase, 'At least 1 symbol in upper case')
    .matches(passValidate.numbers, 'At least 1 number')
    .matches(passValidate.symbols, 'At least 1 service symbol')
    .matches(passValidate.length, 'Length must be more then 8 symbols')
    .required('Required field'),
});
