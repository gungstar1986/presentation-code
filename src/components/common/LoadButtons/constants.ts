export const PREV_BUTTON = 'Load Prev';
export const NEXT_BUTTON = 'Load Next';
