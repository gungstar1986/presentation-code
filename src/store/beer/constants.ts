export const START_PAGE = 0;
export const PER_PAGE = 10;
export const MIN_CONTENT_ITEMS = 1;
export const MAX_CONTENT_ITEMS = 325;

export const paginate = {
  prev: START_PAGE,
  curr: START_PAGE,
  next: START_PAGE,
  per_page: PER_PAGE,
};
