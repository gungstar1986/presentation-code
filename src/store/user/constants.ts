export const EMAIL = 'example@example.com';
export const PASSWORD = 'Test123456*';

export const USER = {
  name: 'Example',
  role: 'User',
  email: EMAIL,
  password: PASSWORD,
};
