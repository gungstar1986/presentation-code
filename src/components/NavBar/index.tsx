import React, { useEffect, useState } from 'react';
import { NavLink, useLocation } from 'react-router-dom';

import { useUserStore } from 'hooks/storesHooks/useUserStore';

import { CABINET, CONTENT, LOGIN } from 'constants/routes';

export const Navbar: React.FC = () => {
  const [location, setLocation] = useState<string>(null);

  const { user } = useUserStore();
  const { pathname } = useLocation();

  const handleLocation = (event: React.MouseEvent, destination: string) => {
    setLocation(pathname);
    if (location.includes(destination)) event.preventDefault();
  };

  useEffect(() => {
    setLocation(pathname);
  }, [pathname]);

  return (
    <div className="main-navbar-container">
      {!user && (
        <div className="navbar-item-container">
          <NavLink onClick={(e) => handleLocation(e, LOGIN)} activeClassName="active-link" to={LOGIN}>
            <span className="navbar-link-text">LOGIN</span>
          </NavLink>
        </div>
      )}

      {user && (
        <>
          <div className="navbar-item-container">
            <NavLink onClick={(e) => handleLocation(e, CABINET)} activeClassName="active-link" to={CABINET}>
              <span className="navbar-link-text">CABINET</span>
            </NavLink>
          </div>
          <div className="navbar-item-container">
            <NavLink onClick={(e) => handleLocation(e, CONTENT)} activeClassName="active-link" to={CONTENT}>
              <span className="navbar-link-text">CONTENT</span>
            </NavLink>
          </div>
        </>
      )}
    </div>
  );
};
