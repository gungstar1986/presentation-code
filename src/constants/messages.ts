// toastify messages
import { EMAIL, PASSWORD } from 'store/user/constants';

export const ERROR_MSG = 'Something went wrong';
export const AUTH_SUCCESS = 'Login successful!';
export const AUTH_ERROR = 'Login or password incorrect';

export const LOGIN_MSG = `Email: ${EMAIL} 
  Password: ${PASSWORD}`;
