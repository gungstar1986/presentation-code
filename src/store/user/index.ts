import { action, makeAutoObservable, observable } from 'mobx';

import { apiUser } from 'api/user';
import { storage } from 'helpers/storage';
import { toastify } from 'helpers/toastify';
import { USER } from 'store/user/constants';
import { User } from 'store/user/interfaces';
import { isUserValid } from 'helpers/isUserValid';
import { AUTH_ERROR, AUTH_SUCCESS, ERROR_MSG } from 'constants/messages';

export class UserStore {
  @observable
  pending = false;

  @observable
  user: User = null;

  @action
  checkUserIsAuth: (user: User) => void = async (user) => {
    try {
      this.pending = true;
      const data = apiUser.userAuth(user);
      this.updateUser({ ...USER, ...data });
    } catch (e) {
      console.error(ERROR_MSG);
    } finally {
      this.pending = false;
    }
  };

  @action
  updateUser: (user: User) => void = (user) => {
    const valid = isUserValid(user);

    if (valid && !this.user) {
      this.user = user;

      toastify.clear();

      return toastify.success(AUTH_SUCCESS);
    }

    return toastify.error(AUTH_ERROR);
  };

  @action
  setUserLogout: () => void = () => {
    this.user = null;
    storage.clear();
  };

  constructor() {
    makeAutoObservable(this);
  }
}

export const userStore = new UserStore();
