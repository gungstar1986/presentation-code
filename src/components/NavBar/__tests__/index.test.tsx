import React from 'react';

import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { Navbar } from 'components/NavBar/index';
import { USER } from 'store/user/constants';

describe('Navbar component', () => {
  beforeAll(() => {
    // @ts-ignore
    React.useContext = () => ({
      userStore: {
        user: USER,
      },
    });
  });

  it('should render Navbar component', () => {
    const component = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(component.baseElement).toMatchSnapshot();
  });
});
