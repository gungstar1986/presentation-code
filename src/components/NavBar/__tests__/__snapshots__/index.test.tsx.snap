// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render Navbar component 1`] = `
<body>
  <div>
    <div
      class="main-navbar-container"
    >
      <div
        class="navbar-item-container"
      >
        <a
          href="/cabinet"
        >
          <span
            class="navbar-link-text"
          >
            CABINET
          </span>
        </a>
      </div>
      <div
        class="navbar-item-container"
      >
        <a
          href="/collection"
        >
          <span
            class="navbar-link-text"
          >
            CONTENT
          </span>
        </a>
      </div>
    </div>
  </div>
</body>
`;
